src/DataShare/DataShare/Account/GetAzDataShareAccount.cs

```diff
@@ -0,0 +1,143 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.DataShare.Account
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+    using System.Net;
+    using Microsoft.Azure.Commands.DataShare.Common;
+    using Microsoft.Azure.Commands.DataShare.Helpers;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.DataShare;
+    using Microsoft.Azure.Management.DataShare.Models;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Extensions;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Properties;
+    using Microsoft.Rest.Azure;
+
+    /// <summary>
+    /// Defines the Get-AzDataShareAccount cmdlet.
+    /// </summary>
+    [Cmdlet(
+         "Get",
+         ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataShareAccount",
+         DefaultParameterSetName = ParameterSetNames.FieldsParameterSet,
+         SupportsPaging = true),
+     OutputType(typeof(PSDataShareAccount))]
+
+    public class GetAzDataShareAccount : AzureDataShareCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of the azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The resource group name of the azure data share account.",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Azure data share account name.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
```

Comment:
@gaurangisaxena do you support the case where the user provides a value `-Name` and not `-ResourceGroupName`? If not, we should refactor the parameter sets so that the user cannot provide a value for `-Name` without also providing a value for `-ResourceGroupName`

Reply:
We do not enforce unique names across subscription, so multiple resource groups can have accounts with the same name. How do you think we should go about handling this?

Reply:
In this case, it's not a matter of if names are unique across a resource group, it's the behavior of the SDK call when `-Name` has a value but `-ResourceGroupName` does not.

Looking at the code path, the following line would be called:

```cs
var account = this.DataShareManagementClient.Accounts.Get(this.ResourceGroupName, this.Name);
```

In this case, `ResourceGroupName` would be `null` and `Name` would have a value, so would this throw an exception, or would it return all the accounts in the subscription with the given name?

If it returns all accounts with that name, then no action is needed, but if it throws an exception, we should split this `FieldsParameterSet`  into two sets:

(1) An optional `-ResourceGroupName` parameter, to allow the user to list all accounts in the subscription or in the specified resource group
(2) Mandatory `-ResourceGroupName` and `-Name` parameters, which forces the user to provide a value for `-ResourceGroupName` if they provide a value for `-Name`